File: src/com/suushiemaniac/cubing/alglib/move/plane/SquareOnePlane.java
package com.suushiemaniac.cubing.alglib.move.plane;

public final class SquareOnePlane implements Plane {
    private final SquareOneHalfPlane halfPlaneUp = SquareOneHalfPlane.UP;
    private final SquareOneHalfPlane halfPlaneDown = SquareOneHalfPlane.DOWN;

    @Override
    public boolean isRotation() {
        return false;
    }

    @Override
    public String toFormatString() {
        return "";
    }
}

File: src/main/kotlin/com/suushiemaniac/cubing/alglib/move/plane/SquareOnePlane.kt
package com.suushiemaniac.cubing.alglib.move.plane

class SquareOnePlane : Plane {
    override val isRotation = false
    override val notation = ""

    override fun equals(other: Any?) = other === INST
    override fun hashCode() = 0

    companion object {
        val INST = SquareOnePlane()
    }
}
